frontend/src/scenes/session-recordings/player/sessionRecordingDataLogic.ts

```diff
@@ -245,7 +245,10 @@ async function processEncodedResponse(
 }
 
 const getSourceKey = (source: SessionRecordingSnapshotSource): string => {
-    return `${source.source}-${source.blob_key}`
+    // realtime sources vary so blob_key is not always present and is either null or undefined...
+    // we only care about key when not realtime
+    // and we'll always have a key when not realtime
+    return `${source.source}-${source.blob_key || source.source}`
```

Comment:
this is one part of the fix here... we ended up with a `realtime-undefined` key and a `realtime-null` key so realtime data was split across two keys when it should be split over one